config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/loader/ServiceLoader.java

```diff
@@ -40,6 +49,65 @@ protected void putToBundle(Bundle bundle, @NotNull Map<String, Service> entities
         bundle.putAllServices(entitiesMap);
     }
 
+    @Override
+    public void load(final Bundle bundle, final File rootDir) {
+        // load services
+        super.load(bundle, rootDir);
+
+        final File policyRootDir = new File(rootDir, "policy");
+        if (!policyRootDir.exists()) {
+            // no policies to bundle. Just return
+            return;
+        } else if (!policyRootDir.isDirectory()) {
+            throw new BundleLoadException("Expected directory but was file: " + policyRootDir);
+        }
+
+        final Map<String, Folder> folderMap = bundle.getFolders();
+        final Folder rootFolder = folderMap.computeIfAbsent(getPath(policyRootDir,policyRootDir), key -> loadFolder(policyRootDir, policyRootDir, null));
+
+        final Map<String, Service> services = bundle.getServices();
+        services.forEach((servicePath, service) -> {
+            int lastFileSeparator = servicePath.lastIndexOf(File.separatorChar);
```

Comment:
Also `FilenameUtils` has some methods that should be helpful here

Reply:
I will look into FilenameUtils to see if this can be neater, thanks!